python/ambassador/VERSION.py

```diff
@@ -13,68 +13,20 @@
 # limitations under the License
 
 import os
-from typing import NamedTuple
+
+Version = "MISSING(FILE)"
+Commit = "MISSING(FILE)"
 
 try:
     # Keep this in-sync with cmd/busyambassador/main.go.
     with open(os.path.join(os.path.dirname(__file__), "..", "ambassador.version")) as version:
-        exec(version.read())
-except FileNotFoundError:
-    BUILD_VERSION = "dirty"
-    GIT_COMMIT = "dirty"
-    GIT_BRANCH = "master"
-    GIT_DIRTY = True
-    GIT_DESCRIPTION = "dirty"
-
-class GitInfo(NamedTuple):
-    commit: str
-    branch: str
-    dirty: bool
-    description: str
-
-
-class BuildInfo(NamedTuple):
-    version: str
-    git: GitInfo
+        info = version.read().split('\n')
+        while len(info) < 2:
+            info.append("MISSING(VAL)")
 
+        Version = info[0]
+        Commit = info[1]
 
-Version = BUILD_VERSION
 
-Build = BuildInfo(
-    version=Version,
-    git=GitInfo(
-        commit=GIT_COMMIT,
-        branch=GIT_BRANCH,
-        dirty=bool(GIT_DIRTY),
-        description=GIT_DESCRIPTION,
-    )
-)
-
-if __name__ == "__main__":
-    import sys
-
-    cmd = "--compact"
-
-    if len(sys.argv) > 1:
-        cmd = sys.argv[1].lower()
-
-    if (cmd == '--version') or (cmd == '-V'):
-        print(Version)
-    elif cmd == '--desc':
-        print(Build.git.description)
-    elif cmd == '--branch':
-        print(Build.git.branch)
-    elif cmd == '--commit':
-        print(Build.git.commit)
-    elif cmd == '--dirty':
-        print(Build.git.dirty)
-    elif cmd == '--all':
-        print("version:         %s" % Version)
-        print("git.branch:      %s" % Build.git.branch)
-        print("git.commit:      %s" % Build.git.commit)
-        print("git.dirty:       %s" % Build.git.dirty)
-        print("git.description: %s" % Build.git.description)
-    else:   # compact
-        print("%s (%s at %s on %s%s)" %
-              (Version, Build.git.description, Build.git.commit, Build.git.branch,
-               " - dirty" if Build.git.dirty else ""))
```

Comment:
I've actually been known to use this...

Reply:
These days you can also `busyambassador version`... it's still a bit different, but it's not a big deal.